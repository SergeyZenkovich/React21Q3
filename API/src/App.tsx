import React, {useEffect, useState} from "react";
import {Switch, Route} from "react-router-dom";
import styled from "styled-components";

import HomePage from "./Components/HomePage/HomePage";
import AboutPage from "./Components/AboutPage/AboutPage";
import Header from "./Components/Header/Header";
import ErrorBlock from "./Components/ErrorComponent/ErrorComponent";
import DetailsPage from "./Components/DetailsPage/DetailsPage";
import {FetchObject} from "./interfaces/ComponentsInterfaces";

const MainComponent = styled.div`
  width: 100%;
  display: flex;
  justify-content: center;
`;

const App = (): JSX.Element => {
  const [activePage, setActivePage] = useState("Home");
  const [pageElements, setPageElements] = useState([] as FetchObject[]);

  return (
    <div>
      <Header activePage={activePage} setActivePage={setActivePage} />
      <MainComponent>
        <Switch>
          <Route exact path="/about" render={() => <AboutPage setActivePage={setActivePage} />} />
          <Route
            exact
            path="/details/:id"
            render={() => <DetailsPage setActivePage={setActivePage} />}
          />
          <Route
            exact
            path="/"
            render={() => (
              <HomePage setActivePage={setActivePage} setPageElements={setPageElements} />
            )}
          />
          <Route path="/*" render={() => <ErrorBlock />} />
        </Switch>
      </MainComponent>
    </div>
  );
};
export default App;
