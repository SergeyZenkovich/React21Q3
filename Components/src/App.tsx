import React from "react";
import styled from "styled-components";
import Card from "./Components/Card/Card";
import Serach from "./Components/SearchBar/SearchBar";
import cardsData from "./Data";
import {CardInterface} from "./interfaces/ComponentsInterfaces";

const StyledCardsBlock = styled.div`
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
`;

const App = (): JSX.Element => {
  const Cards = cardsData.map((el: CardInterface) => (
    <Card
      key={el.cardName}
      cardName={el.cardName}
      cardPrice={el.cardPrice}
      cardYear={el.cardYear}
      cardLikes={el.cardLikes}
      cardRating={el.cardRating}
      cardImg={el.cardImg}
    />
  ));
  return (
    <div>
      <Serach />
      <StyledCardsBlock>{Cards}</StyledCardsBlock>
    </div>
  );
};
export default App;
