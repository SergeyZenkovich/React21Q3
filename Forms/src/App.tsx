import React, {useState} from "react";
import styled from "styled-components";
import Card from "./Components/Card/Card";
import Form from "./Components/Form/Form";
import Serach from "./Components/SearchBar/SearchBar";
import cardsData from "./Data";
import {CardInterface} from "./interfaces/ComponentsInterfaces";

const StyledCardsBlock = styled.div`
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
`;

const App = (): JSX.Element => {
  const [cards, setCards] = useState([] as CardInterface[]);
  const addCard = (newCard: CardInterface): void => {
    setCards([...cards, newCard]);
  };
  const Cards = cards.map((el: CardInterface) => (
    <Card
      key={el.cardProblem}
      cardProblem={el.cardProblem}
      cardDate={el.cardDate}
      cardPriority={el.cardPriority}
      cardisRequestNeeded={el.cardisRequestNeeded}
    />
  ));
  return (
    <div>
      <Form addCard={addCard} />
      <StyledCardsBlock>{Cards}</StyledCardsBlock>
    </div>
  );
};
export default App;
