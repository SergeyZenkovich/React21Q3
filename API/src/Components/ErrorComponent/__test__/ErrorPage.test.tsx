import React, {useState} from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import {act} from "react-dom/test-utils";
import ErrorBlock from "../ErrorComponent";
import App from "../../../App";

const testString = "Car";

it("renders Error page without crashing", () => {
  const container = document.createElement("div");
  ReactDOM.render(<ErrorBlock query={testString}></ErrorBlock>, container);

  expect(container.querySelector("h4").textContent).toBe("Sorry no results found for Car");
});
it("renders Error page without query", () => {
  const container = document.createElement("div");
  ReactDOM.render(<ErrorBlock></ErrorBlock>, container);

  expect(container.querySelector("h4").textContent).toBe("Sorry page not found ");
});

describe("AboutPage Component", () => {
  const setFunction = jest.fn();
  let container;
  beforeEach(() => {
    container = renderer.create(<ErrorBlock query={testString}></ErrorBlock>);
  });
  it("matches shapshot", () => {
    const tree = container.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
