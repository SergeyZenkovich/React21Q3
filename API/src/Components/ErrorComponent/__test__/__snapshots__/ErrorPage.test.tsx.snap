// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutPage Component matches shapshot 1`] = `
<div
  className="sc-bdnxRM dYoKXS"
>
  <h4>
    Sorry no results found for 
    Car
  </h4>
  <img
    alt="no results img"
    src="test-file-stub"
  />
</div>
`;
